Permissions/AutorizationProfiles/Retrieve.ps1

```diff
@@ -0,0 +1,186 @@
+                SkipToken = $Null
+            }
+        }
+
+        do {
+            $Result = Invoke-RestMethod @LisaRequest @SplatParams
+
+            $SplatParams.Body.SkipToken = $Result.nextLink
+
+            if ($Result -is [array]) {
+                Write-Output $Result
+            }
+            else {
+                Write-Output $Result.value
+            }
+        }
+        until([string]::IsNullOrWhiteSpace($Result.nextLink))
+    }
+    catch {
+        $PSCmdlet.ThrowTerminatingError($PSItem)
+    }
+}
+
+
+function Resolve-ErrorMessage {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory, ValueFromPipeline)]
+        [object]
+        $ErrorObject
+    )
+
+    process {
+        $Exception = [PSCustomObject]@{
+            FullyQualifiedErrorId = $ErrorObject.FullyQualifiedErrorId
+            MyCommand             = $ErrorObject.InvocationInfo.MyCommand
+            RequestUri            = $ErrorObject.TargetObject.RequestUri
+            ScriptStackTrace      = $ErrorObject.ScriptStackTrace
+            ErrorMessage          = $Null
+            VerboseErrorMessage   = $Null
+        }
+
+        switch ($ErrorObject.Exception.GetType().FullName) {
+            "Microsoft.PowerShell.Commands.HttpResponseException" {
+                $Exception.ErrorMessage = $ErrorObject.ErrorDetails.Message
+                break
+            }
+            "System.Net.WebException" {
+                $Exception.ErrorMessage = [System.IO.StreamReader]::new(
+                    $ErrorObject.Exception.Response.GetResponseStream()).ReadToEnd()
+                break
+            }
+            default {
+                $Exception.ErrorMessage = $ErrorObject.Exception.Message
+            }
+        }
+
+        $Exception.VerboseErrorMessage = @(
+            "Error at Line [$($ErrorObject.InvocationInfo.ScriptLineNumber)]: $($ErrorObject.InvocationInfo.Line)."
+            "ErrorMessage: $($Exception.ErrorMessage) [$($ErrorObject.ErrorDetails.Message)]"
+        ) -Join ' '
+
+        Write-Output $Exception
+    }
+}
+#endregion functions
+
+
+#region Aliasses
+$Config = $ActionContext.Configuration
+#endregion Aliasses
+
+
+# Start Script
+try {
+    # Getting accessToken
+    $AccessToken = $Config.AzureAD | Get-LisaAccessToken -AsSecureString
+
+    $SplatParams = @{
+        Uri         = $Config.BaseUrl
+        Endpoint    = "AuthorizationProfiles"
+        AccessToken = $AccessToken
+    }
+    $AuthorizationProfiles = Get-LisaCollection @SplatParams
+
+    $OutputContext.Permissions = $AuthorizationProfiles | ForEach-Object {
+        [PSCustomObject]@{
+            DisplayName    = "AuthorizationProfile - $($PSItem.displayName)"
+            Identification = @{
+                Reference = $PSItem.id
+            }
+        }
+    }
+}
+catch {
+    $Exception = $PSItem | Resolve-ErrorMessage
+
+    Write-Verbose -Verbose $Exception.VerboseErrorMessage
+
+    throw $Exception.ErrorMessage
+}
```

Comment:
In our best practices we use a try, catch, finally
in the finally part we put everything into the outputcontext. 
make sure it is in the following syntax: $outputContext.Permissions.Add()
here is a short example:
```powershell
finally {
    # Send results
    $m365Groups | ForEach-Object {
        # Shorten DisplayName to max. 100 chars
        $displayName = "Microsoft 365 Group - $($_.displayName)"
        $displayName = $displayName.substring(0, [System.Math]::Min(100, $displayName.Length)) 
        
        $outputContext.Permissions.Add(
            @{
                displayName    = $displayName
                identification = @{
                    Id   = $_.id
                    Name = $_.displayName
                }
            }
        )
    }
}
```

Reply:
I don´t see how this can be best practice. Essentially you say that if something goes wrong, we will still try to return permissions. 

A finally block should be used to clean up everything that needs to be cleaned up, like a remote session, and should not be used as a way to continue the script.

Reply:
agreed, I believe they did this because this part is about retrieval of the permissions. Not actually an action for the auditlogs